lib/methods/import.js

```diff
@@ -232,12 +232,49 @@ module.exports = self => {
       return locale;
     },
 
-    async insertDocs(req, docs, reporting) {
+    async insertDocs(req, docs, moduleName, reporting) {
       const duplicatedDocs = [];
       const duplicatedIds = [];
       const failedIds = [];
 
       for (const doc of docs) {
+        const { updateKey, updateField } = self.getUpdateKey(doc);
+
+        // If an update key is found, we try to update the document.
+        // It also means that we are in a "simple" import (CSV or Excel),
+        // not in a full import process like Gzip with JSON formats.
+        if (updateKey) {
```

Comment:
if an update key is found, we call `self.insertOrUpdateDocWithKey` rather than `self.insertOrUpdateDoc` because it has its own logic.